src/jobs/models/database/sequelize/sequelizeConnector.js

```diff
@@ -83,23 +91,41 @@ async function getJob(jobId) {
     return allJobs;
 }
 
+async function getJobType(jobId) {
+    const jobs = await getJob(jobId);
+    return jobs[0].type;
+}
+
 async function updateJob(jobId, jobInfo) {
     const job = client.model('job');
 
-    let params = {
+    const params = {
         test_id: jobInfo.test_id,
-        arrival_rate: jobInfo.arrival_rate,
+        type: jobInfo.type,
         cron_expression: jobInfo.cron_expression,
         duration: jobInfo.duration,
         environment: jobInfo.environment,
-        ramp_to: jobInfo.ramp_to,
         parallelism: jobInfo.parallelism,
         max_virtual_users: jobInfo.max_virtual_users,
         proxy_url: jobInfo.proxy_url,
         debug: jobInfo.debug,
         enabled: jobInfo.enabled
     };
 
+    const jobType = jobInfo.type || await getJobType(jobId);
+    switch (jobType) {
+    case JOB_TYPE_FUNCTIONAL_TEST:
+        params.arrival_count = jobInfo.arrival_count;
+        params.arrival_rate = null;
+        params.ramp_to = null;
+        break;
+    case JOB_TYPE_LOAD_TEST:
+        params.arrival_rate = jobInfo.arrival_rate;
+        params.ramp_to = jobInfo.ramp_to;
+        params.arrival_count = null;
+        break;
+    }
+
```

Comment:
I am a big fan of having a `case default`, it can just throw an error + I really like to put curly braces around each `case`, totally up to you

Reply:
not sure i want to handle an unknown case and throw an error, it should not get to that stage and means data in the db is corrupt. let's leave it like this

Reply:
@NivLipetz  what do you think?

Reply:
agreed, will fix ✌️